pkg/deviceshifu/mockdevice/mockdevice-robot-arm/mockdevice-robot-arm_test.go

```diff
@@ -0,0 +1,73 @@
+package main
+
+import (
+	"github.com/edgenesis/shifu/pkg/deviceshifu/mockdevice/mockdevice"
+	"github.com/stretchr/testify/assert"
+	"io"
+	"net/http"
+	"os"
+	"strings"
+	"testing"
+	"time"
+)
+
+func TestInstructionHandler(t *testing.T) {
+	availableFuncs := []string{
+		"get_coordinate",
+		"get_status",
+	}
+	os.Setenv("MOCKDEVICE_NAME", "mockdevice_test")
+	os.Setenv("MOCKDEVICE_PORT", "12345")
+	mocks := []struct {
+		name       string
+		url        string
+		StatusCode int
+		expResult  interface{}
+	}{
+		{
+			"case 1 port 12345 get_coordinate",
+			"http://localhost:12345/get_coordinate",
+			200,
+			true,
+		},
+		{
+			"case 2 port 12345 get_status",
+			"http://localhost:12345/get_status",
+			200,
+			[]string{"Running", "Idle", "Busy", "Error"},
+		},
+	}
+
+	go mockdevice.StartMockDevice(availableFuncs, instructionHandler)
+
+	time.Sleep(100 * time.Microsecond)
+
+	for _, c := range mocks {
+		t.Run(c.name, func(t *testing.T) {
+			resp, err := http.Get(c.url)
+			assert.Nil(t, err)
+			defer resp.Body.Close()
+			body, _ := io.ReadAll(resp.Body)
+
+			switch {
+			case strings.Contains(c.url, "/get_coordinate"):
+				assert.Equal(t, c.expResult, check([]string{"xpos", "ypos", "zpos"}, string(body)))
+			case strings.Contains(c.url, "/get_status"):
+				assert.Contains(t, c.expResult, string(body))
+			}
+
+		})
+	}
+
+}
+
+func check(expResult interface{}, Result string) bool {
```

Comment:
try to make the function using narrow type

func checkContains(expResult []string, Result string) bool